### ISSUE TITLE
Bulder.update() support

### ISSUE THREAD

**Issue by tomas-langer** [oracle/helidon#1788]
Currently when we need to update the builder in another method, we need to do something like this:
```java
SomeClass.Builder builder = SomeClass.builder();
updateBuilder(builder);
SomeClass instance = builder.addSomething("something")
        .build();
```
This break in fluent API and the requirement to create a local variable could be avoided if we add the following method to `io.helidon.common.Builder` and change the type configuration of it (and all builders implementing it):
```java
    default B update(Consumer<B> consumer) {
        consumer.accept((B)this);
        return (B) this;
    }
```

Then the original code could be refactored to:
```java
SomeClass instance = SomeClass.builder()
        .update(this::updateBuilder)      
        .addSomething("something")
        .build();
```

Which seems to be much cleaner.

This change should be possible without breaking backward compatibility - only if somebody implements our `io.helidon.common.Builder`, they would need to add a type argument.

**Comment by tomas-langer** [comment]
This would be the type declaration:
`public interface Builder<B extends Builder<B, T>, T> extends Supplier<T> `

**Comment by tomas-langer** [comment]
And this would be type declaration of an implementation:
`class MyBuilder implements Builder<MyBuilder, MyClass>`

**Comment by tomas-langer** [comment]
Merged as part of preparation for Helidon 3.0.0
